### PR TITLE
 feat(content-sidebar): Handle additional tab image failures

### PR DESCRIPTION
**Pull request by jeremypress**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jeremypress has signed the CLA. Thanks for the pull request!

File: src/components/avatar/Avatar.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ class Avatar extends React.PureComponent<Props, State> {
     };
 
     render() {
-        const { avatarUrl, className, name, id, size }: Props = this.props;
+        const { avatarUrl, className, name, id, size = '' }: Props = this.props;
```
Comment (jeremypress) [inline src/components/avatar/Avatar.js]: flow fix

File: src/elements/content-sidebar/additional-tabs/AdditionalTab.js [hunk 2]
```diff
@@ -18,20 +18,51 @@ type Props = {
...
+    render() {
+        const { callback: callbackFn, id, isLoading, iconUrl, onImageLoad, title, ...rest } = this.props;
+        const { isErrored } = this.state;
+        const className = classNames('bdl-AdditionalTab', { 'bdl-AdditionalTab--hidden': isLoading });
+
+        let TabIcon;
+
+        if (isErrored) {
+            TabIcon = <AdditionalTabPlaceholder isLoading={false} />;
+        } else if (id && id > 0 && iconUrl) {
```
Comment (JustinHoldstock) [inline src/elements/content-sidebar/additional-tabs/AdditionalTab.js]: Following this logic, if the user supplies `undefined` or `''` as an `iconUrl`, it'll display the  `IconEllipsis`.

Is this intended, or should we have a missing url icon/use the `AdditionalTabPlaceholder`?

File: src/elements/content-sidebar/additional-tabs/AdditionalTabPlaceholder.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/**
+ * @flow
+ * @file Preview sidebar additional tab loading component
+ * @author Box
+ */
+
+import classNames from 'classnames';
+import * as React from 'react';
```
Comment (jstoffan) [inline src/elements/content-sidebar/additional-tabs/AdditionalTabPlaceholder.js]: React should usually come first in the import list.

File: src/elements/content-sidebar/additional-tabs/AdditionalTabPlaceholder.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+type Props = {
+    isLoading: boolean,
+};
+
+const AdditionalTabPlaceholder = ({ isLoading = false }: Props) => {
+    const classes = classNames('bdl-AdditionalTab-icon', 'bdl-AdditionalTabLoading-placeholder', {
```
Comment (jstoffan) [inline src/elements/content-sidebar/additional-tabs/AdditionalTabPlaceholder.js]: Should this use `bdl-AdditionalTab-placeholderIcon`, instead?

File: src/elements/content-sidebar/additional-tabs/AdditionalTab.scss [hunk 5]
```diff
@@ -14,7 +14,8 @@
     }
 }
 
-.bdl-AdditionalTab-icon {
+.bdl-AdditionalTab-icon,
+.bdl-AdditionalTab-placeholderIcon {
```
Comment (jstoffan) [inline src/elements/content-sidebar/additional-tabs/AdditionalTab.scss]: Is this unused?

File: src/elements/content-sidebar/additional-tabs/AdditionalTabPlaceholder.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+/**
+ * @flow
+ * @file Preview sidebar additional tab loading component
```
Comment (jstoffan) [inline src/elements/content-sidebar/additional-tabs/AdditionalTabPlaceholder.js]: Does the comment here need to be updated?

File: src/elements/content-sidebar/additional-tabs/AdditionalTab.js [hunk 7]
```diff
@@ -18,20 +18,51 @@ type Props = {
...
+    isErrored: boolean,
 };
 
+class AdditionalTab extends React.PureComponent<Props, State> {
+    state = {
+        isErrored: false,
+    };
+
+    onImageError = () => {
+        this.props.onImageLoad();
```
Comment (jstoffan) [inline src/elements/content-sidebar/additional-tabs/AdditionalTab.js]: Is the `onImageLoad` callback expecting any arguments? Is there anything passed from the onload event?

File: src/elements/content-sidebar/additional-tabs/AdditionalTabPlaceholder.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+type Props = {
+    isLoading: boolean,
+};
+
+const AdditionalTabPlaceholder = ({ isLoading = false }: Props) => {
+    const classes = classNames('bdl-AdditionalTab-icon', 'bdl-AdditionalTabLoading-placeholder', {
+        'bdl-AdditionalTabLoading-placeholder--loading': isLoading,
+    });
+    return (
+        <div className="bdl-AdditionalTabLoading">
```
Comment (jstoffan) [inline src/elements/content-sidebar/additional-tabs/AdditionalTabPlaceholder.js]: Should this be `bdl-AdditionalTabPlaceholder`?

File: src/elements/content-sidebar/additional-tabs/AdditionalTab.js (hunk 7 above)
Comment (jeremypress) [reply src/elements/content-sidebar/additional-tabs/AdditionalTab.js]: onImageLoad adds to the global count of loaded images, so it doesn't need any specifics from an individual load. Based on the MDN example I was looking at they don't expect any arguments to come back from that event.

File: src/elements/content-sidebar/additional-tabs/AdditionalTab.scss [hunk 9]
```diff
@@ -14,17 +14,16 @@
...
     @include bdl-fadeIn(.5s);
 
     height: 24px;
     width: 24px;
 }
 
-.bdl-AdditionalTab-moreIcon .fill-color {
-    fill: $downtown-grey;
-}
+// .bdl-AdditionalTab-moreIcon .fill-color {
```
Comment (jeremypress) [inline src/elements/content-sidebar/additional-tabs/AdditionalTab.scss]: removing this.

**Pull request merged**